### PR TITLE
Change Nullable to default to enable in libraries src and ref projects

### PR DESCRIPTION
**Pull request by eerhardt**
Now that 90% of our libraries are annotated for nulllable, it makes sense to have the default be `enable`. There are currently 26 libraries that aren't fully annotated yet:

```
Microsoft.Bcl.AsyncInterfaces
Microsoft.Extensions.DependencyInjection.Specification.Tests
Microsoft.Extensions.Hosting.Systemd
Microsoft.Extensions.Hosting.WindowsServices
Microsoft.NETCore.Platforms
Microsoft.XmlSerializer.Generator
System.CodeDom
System.ComponentModel.Composition.Registration
System.Composition
System.Composition.AttributedModel
System.Composition.Convention
System.Composition.Hosting
System.Composition.Runtime
System.Composition.TypedParts
System.Configuration.ConfigurationManager
System.Diagnostics.EventLog
System.Diagnostics.PerformanceCounter
System.DirectoryServices.AccountManagement
System.DirectoryServices.Protocols
System.IO.Ports
System.Management
System.Runtime.Caching
System.Security.Cryptography.Xml
System.Security.Permissions
System.ServiceModel.Syndication
System.Speech
```

### CODE REVIEW

**Review by jeffhandley — APPROVED** [review]

File: src/libraries/Directory.Build.props [hunk 1]
```diff
@@ -96,8 +96,8 @@
     <GenerateDocumentationFile Condition="'$(IsSourceProject)' == 'true' and '$(MSBuildProjectExtension)' != '.vbproj'">true</GenerateDocumentationFile>
     <CLSCompliant Condition="'$(CLSCompliant)' == '' and '$(IsTestProject)' != 'true' and '$(IsTestSupportProject)' != 'true'">true</CLSCompliant>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(IsSourceProject)' == 'true'">
-    <!-- Nullability is enabled by default for all src projects. -->
+  <PropertyGroup Condition="'$(IsSourceProject)' == 'true' or '$(IsReferenceAssemblyProject)' == 'true'">
+    <!-- Nullability is enabled by default for all src and ref projects. -->
     <Nullable Condition="'$(Nullable)' == ''">enable</Nullable>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.props:101]: Do we only want to set this property if it isn't yet set? Why not set it unconditionally? What if another component like the SDK or Arcade sets that property in the future and with that overrides our setting?

File: src/libraries/System.CodeDom/ref/System.CodeDom.csproj [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppMinimum);netstandard2.0;$(NetFrameworkMinimum)</TargetFrameworks>
+    <Nullable>disable</Nullable>
```
Comment (ViktorHofer) [inline src/libraries/System.CodeDom/ref/System.CodeDom.csproj:4]: Do we have a tracking issue to enable nullability for projects like CodeDom or don't we plan to do so?

**Review by ViktorHofer — APPROVED** [review]
LGTM except for the few questions. Thanks Eric

File: src/libraries/System.CodeDom/ref/System.CodeDom.csproj (hunk 2 above)
Comment (eerhardt) [reply src/libraries/System.CodeDom/ref/System.CodeDom.csproj:4]: Yes, https://github.com/dotnet/runtime/issues/41720

File: src/libraries/Directory.Build.props (hunk 1 above)
Comment (ViktorHofer) [reply src/libraries/Directory.Build.props:101]: I usually unconditionally overwrite properties / items if I want to set a (new) default regardless of what was previously set. In this case it sounds like we want to set our own default. That said, I don't have a strong opinion, so I'm fine leaving it as is if you prefer that.

**Comment by eerhardt** [comment]
runtime-dev-innerloop (Build OSX x64 release Runtime_Debug) timing out is unrelated to this change.

Merging.

**Pull request merged**